File: evenOrOdd.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <vector>

using namespace std;

void checkNumbers(string mode){
    if (mode == "list"){
        int numberOfEvens = 0;
        int numberOfOdds = 0;
        while (true){
            cout << "\nEnter numbers separated by spaces: ";
            string input;
            getline(cin, input);
            stringstream ss(input);
            string num;
            vector<string> results;
            bool valid_input = true;
            float val;
            while (ss >> num){
                try {
                    val = stof(num);
                } catch (invalid_argument&) {
                    cout << "\nInvalid input. Please enter numbers only." << endl;
                    valid_input = false;
                    break;
                }
                if (static_cast<int>(val) % 2 == 0){
                    results.push_back("Even");
                } else {
                    results.push_back("Odd");
                }
            }
            if (valid_input){
                for (const string& result : results){
                    if (result == "\nEven"){
                        numberOfEvens++;
                    } else {
                        numberOfOdds++;
                    }
                }
                break;
            }
        }
        cout << "\nNumber of evens: " << numberOfEvens << endl;
        cout << "\nNumber of odds: " << numberOfOdds << endl;
    } else if (mode == "single"){
        while (true){
            cout << "\nEnter a number: ";
            string num;
            getline(cin, num);
            try {
                float val = stof(num);
                if (static_cast<int>(val) % 2 == 0){
                    cout << "\nEven" << endl;
                } else {
                    cout << "\nOdd" << endl;
                }
                break;
            } catch (invalid_argument&) {
                cout << "\nInvalid input. Please enter a number." << endl;
            }
        }
    }
}

int main() {
    while (true) {
        string listOrSingle;
        cout << "\nEnter 'list' to check multiple numbers or 'single' to check one number: ";
        getline(cin, listOrSingle);
        checkNumbers(listOrSingle);

        string again;
        cout << "\nDo you want to check again? (yes/no): ";
        getline(cin, again);
        if (again != "yes" && again != "y") {
            break; // exit loop and program
        }
    }
    return 0;
}

File: evenOrOdd.py
def check_numbers(mode):
    if mode == 'list':
        numberOfEvens = 0
        numberOfOdds = 0
        while True:
            numbers = input("\nEnter numbers separated by spaces: ").strip().split()
            results = []
            valid_input = True
            for num in numbers:
                try:
                    val = float(num)
                except ValueError:
                    print("\nInvalid input. Please enter numbers only.")
                    valid_input = False
                    break
                if int(val) % 2 == 0:
                    results.append("\nEven")
                else:
                    results.append("\nOdd")
            if valid_input:
                for result in results:
                    if result == "\nEven":
                        numberOfEvens += 1
                    else:
                        numberOfOdds += 1
                break
        print(f"\nNumber of evens: {numberOfEvens}")
        print(f"\nNumber of odds: {numberOfOdds}")
    elif mode == 'single':
        while True:
            num = input("\nEnter a number: ").strip()
            try:
                val = float(num)
                if int(val) % 2 == 0:
                    print("\nEven")
                else:
                    print("\nOdd")
                break
            except ValueError:
                print("\nInvalid input. Please enter a number.")


while True:
    listOrSingle = input("\nEnter 'list' to check multiple numbers or 'single' to check one number: ").strip().lower()
    check_numbers(listOrSingle)
    again = input("\nDo you want to check again? (yes/no): ").strip().lower()
    if again != 'yes' and again != 'y':
        break